trankill1127/w07/BOJ_1238.java

```diff
@@ -0,0 +1,94 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.PriorityQueue;
+import java.util.StringTokenizer;
+
+public class BOJ_1238 {
+
+    public static class Road implements Comparable<Road>{
+        int v;
+        int time;
+
+        Road(int v, int time){
+            this.v=v;
+            this.time=time;
+        }
+
+        @Override
+        public int compareTo(Road o) {
+            return this.time-o.time;
+        }
+    }
+
+    public static void main(String[] args) throws IOException {
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        StringTokenizer st = new StringTokenizer(br.readLine(), " ");
+        n = Integer.parseInt(st.nextToken()); //인원 수
+        int m = Integer.parseInt(st.nextToken()); //경로의 수
+        x = Integer.parseInt(st.nextToken()); //목적지 마을
+
+        map = new ArrayList<>();
+        for (int i=0; i<=n;i++) map.add(new ArrayList<>());
+        for (int i=0; i<m; i++){
+            st = new StringTokenizer(br.readLine(), " ");
+            int s = Integer.parseInt(st.nextToken());
+            int e = Integer.parseInt(st.nextToken());
+            int t = Integer.parseInt(st.nextToken());
+            map.get(s).add(new Road(e,t));
+        }
+
+        goTime = new int[n+1];
+        backTime = new int[n+1];
+
+        for (int i=1; i<=n; i++){
+            if (i==x) continue;
+            dajikstra(i,x);
+        }
+        dajikstra(x,0);
+
+        int maxSumTime=0;
+        for (int i=1; i<=n; i++){
+            int sum = goTime[i]+backTime[i];
+            if (maxSumTime<sum) maxSumTime=sum;
+        }
+        System.out.println(maxSumTime);
+    }
+
+    public static int n;
+    public static int x;
+    public static ArrayList<ArrayList<Road>> map;
+    public static int[] goTime;
+    public static int[] backTime;
+
+    public static void dajikstra(int start, int goal){
+        int[] timeSum = new int[n+1];
+        Arrays.fill(timeSum, Integer.MAX_VALUE);
```

Comment:
timeSum 배열이 `new` 연산을 통해 n + 1 크기 만큼 새롭게 메모리를 할당받는 코드네요.
매번 다익스트라 함수가 호출 될 때마다, 메모리를 할당받기 때문에, 메모리를 많이 잡아먹을 것 같아요!
이럴 때는 `timeSum` 배열을 지역 변수로 정의할 필요는 없어요! 
`timeSum`을 static 변수로 선언 한 다음, 값만 초기화 해주시면 됩니다~
```suggestion
    public static int n;
    public static int x;
    public static ArrayList<ArrayList<Road>> map;
    public static int[] goTime;
    public static int[] backTime;
    public static int[] timeSum; // 초기 선언 & 메모리 할당은 이미 됐다고 가정

    public static void dajikstra(int start, int goal){
        Arrays.fill(timeSum, Integer.MAX_VALUE);
```

Reply:
새롭게 할당받으면 기존에 들어있던 값들은 자동으로 메모리에서 삭제되는 줄 알았는데 아니었군요!ㅠㅠㅠㅠ
오개념을 바로잡게 해줘서 고마워요 따봉병익🤓